server/src/main/java/io/druid/guice/http/HttpClientModule.java

```diff
@@ -118,7 +118,10 @@ public HttpClient get()
         builder.withSslContext(getSslContextBinding().getProvider().get());
       }
 
-      return HttpClientInit.createClient(builder.build(), LifecycleUtils.asMmxLifecycle(getLifecycleProvider().get()));
+      return HttpClientInit.createClient(
```

Comment:
maybe we can do something here so that httpClient is wrapped with appropriate internal authenticator.